packages/backend-data/src/factory.ts

```diff
@@ -111,16 +117,30 @@ class DataGenerator implements ConstructContainerEntryGenerator {
     scope,
     ssmEnvironmentEntriesGenerator,
   }: GenerateContainerEntryProps) => {
-    let amplifyGraphqlDefinition;
-    let jsFunctions: JsResolver[] = [];
-    let functionSchemaAccess: FunctionSchemaAccess[] = [];
-    let lambdaFunctions: Record<string, ConstructFactory<AmplifyFunction>> = {};
+    const amplifyGraphqlDefinitions: IAmplifyDataDefinition[] = [];
+    const schemasJsFunctions: JsResolver[] = [];
+    const schemasFunctionSchemaAccess: FunctionSchemaAccess[] = [];
+    let schemasLambdaFunctions: Record<
+      string,
+      ConstructFactory<AmplifyFunction>
+    > = {};
     try {
-      if (isModelSchema(this.props.schema)) {
-        ({ jsFunctions, functionSchemaAccess, lambdaFunctions } =
-          this.props.schema.transform());
-      }
-      amplifyGraphqlDefinition = convertSchemaToCDK(this.props.schema);
+      const schemas = isCombinedSchema(this.props.schema)
+        ? this.props.schema.schemas
+        : [this.props.schema];
+      schemas.forEach((schema) => {
+        if (isModelSchema(schema)) {
+          const { jsFunctions, functionSchemaAccess, lambdaFunctions } =
+            schema.transform();
+          schemasJsFunctions.push(...jsFunctions);
+          schemasFunctionSchemaAccess.push(...functionSchemaAccess);
+          schemasLambdaFunctions = {
+            ...schemasLambdaFunctions,
+            ...lambdaFunctions,
+          };
```

Comment:
I thought about pushing the surrounding logic into a helper, but we're merging locals with locals so its a bigger refactor than seemed worth it before getting a bit of external input.